Added starred documents screen and starring functionality

Saransh-cpp opened the pull request:
Closes #131 
Addressing #99 

## Additions
1. A new Hive box to store all the starred documents
2. Starred documents screen (similar to the home screen)
3. Option for starring from home screen (in every PDF's card widget)
4. Sharing option from the starred screen

## Things left to add
The un-starring option is something that needs to be implemented. It will be very similar to the delete option, but rather than deleting from ```pdfs``` box, the PDF will be deleted from ```starred``` box.
I have been trying but I haven't figured out how to delete the PDFs yet. Both these deleting features can be added in a single PR. After this PR is merged, I will open a separate issue for the delete function so that if someone can implement that, he/she can go ahead.

Please let me know if you want any changes in this PR. Thanks!

(MWoC and CWoC participant)

Saransh-cpp committed:
Added starred documents' Hive box

Saransh-cpp committed:
Added logic for starring a document

Saransh-cpp committed:
Added starred document screen

Saransh-cpp committed:
Added route

Saransh-cpp committed:
Fixed issue with duplicate starred documents

Saransh-cpp committed:
Added snackbar

anushbhatia reviewed (changes requested):
![image](https://user-images.githubusercontent.com/40017559/108886602-2ae31e00-762f-11eb-98ec-384a0024e7fa.png)

https://user-images.githubusercontent.com/40017559/108886621-320a2c00-762f-11eb-9c9d-4b795a86528e.mp4
Change the star colour on staring it
Resolve the overflow here

Saransh-cpp commented:
> ![image](https://user-images.githubusercontent.com/40017559/108886602-2ae31e00-762f-11eb-98ec-384a0024e7fa.png)
> 
> https://user-images.githubusercontent.com/40017559/108886621-320a2c00-762f-11eb-9c9d-4b795a86528e.mp4
> Change the star colour on staring it
> Resolve the overflow here

I will make the necessary changes!

Saransh-cpp committed:
Made the star icon dynamic in Home.dart

Saransh-cpp committed:
Fixed renderflex overflow issue when PDF is created

Saransh-cpp commented:
## Changes
1. Added a function to check if the PDF's path is stored in ```starred``` Hive Box, if it is then the function returns true and changes the empty start icon to filled star icon.
2. The overflow was caused because the keyboard in PDFConversion screen was not being discarded automatically, added a line to discard the keyboard.

Please let me know if any further changes are needed, thanks!
